src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-multisite-migrate/rgw-multisite-migrate.component.ts

```diff
@@ -0,0 +1,196 @@
+import { NotificationType } from '~/app/shared/enum/notification-type.enum';
+import { CdFormGroup } from '~/app/shared/forms/cd-form-group';
+import { CdValidators } from '~/app/shared/forms/cd-validators';
+import { NotificationService } from '~/app/shared/services/notification.service';
+import { RgwRealm, RgwZone, RgwZonegroup } from '../models/rgw-multisite';
+import { ModalService } from '~/app/shared/services/modal.service';
+
+@Component({
+  selector: 'cd-rgw-multisite-migrate',
+  templateUrl: './rgw-multisite-migrate.component.html',
+  styleUrls: ['./rgw-multisite-migrate.component.scss']
+})
+export class RgwMultisiteMigrateComponent implements OnInit {
+  readonly endpoints = /^((https?:\/\/)|(www.))(?:([a-zA-Z]+)|(\d+\.\d+.\d+.\d+)):\d{2,4}$/;
+  readonly ipv4Rgx = /^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/i;
+  readonly ipv6Rgx = /^(?:[a-f0-9]{1,4}:){7}[a-f0-9]{1,4}$/i;
+
+  @Output()
+  submitAction = new EventEmitter();
+
+  multisiteMigrateForm: CdFormGroup;
+  zoneNames: string[];
+  realmList: RgwRealm[];
+  multisiteInfo: object[] = [];
+  realmNames: string[];
+  zonegroupList: RgwZonegroup[];
+  zonegroupNames: string[];
+  zoneList: RgwZone[];
+  realm: RgwRealm;
+  zonegroup: RgwZonegroup;
+  zone: RgwZone;
+  newZonegroupName: any;
+  newZoneName: any;
+  bsModalRef: NgbModalRef;
+
+  constructor(
+    public activeModal: NgbActiveModal,
+    public actionLabels: ActionLabelsI18n,
+    public rgwMultisiteService: RgwMultisiteService,
+    public rgwZoneService: RgwZoneService,
+    public notificationService: NotificationService,
+    public rgwZonegroupService: RgwZonegroupService,
+    public rgwRealmService: RgwRealmService,
+    public modalService: ModalService
+  ) {
+    this.createForm();
+  }
+
+  createForm() {
+    this.multisiteMigrateForm = new CdFormGroup({
+      realmName: new FormControl(null, {
+        validators: [
+          Validators.required,
+          CdValidators.custom('uniqueName', (realmName: string) => {
+            return this.realmNames && this.zoneNames.indexOf(realmName) !== -1;
+          })
+        ]
+      }),
+      zonegroupName: new FormControl(null, {
+        validators: [
+          Validators.required,
+          CdValidators.custom('uniqueName', (zonegroupName: string) => {
+            return this.zonegroupNames && this.zoneNames.indexOf(zonegroupName) !== -1;
+          })
+        ]
+      }),
+      zoneName: new FormControl(null, {
+        validators: [
+          Validators.required,
+          CdValidators.custom('uniqueName', (zoneName: string) => {
+            return this.zoneNames && this.zoneNames.indexOf(zoneName) !== -1;
+          })
+        ]
+      }),
+      zone_endpoints: new FormControl([], {
+        validators: [
+          CdValidators.custom('endpoint', (value: string) => {
+            if (_.isEmpty(value)) {
+              return false;
+            } else {
+              if (value.includes(',')) {
```

Comment:
Zone endpoints is marqued as a required field, but it doesn't seem to be required

![image](https://user-images.githubusercontent.com/106810838/237039326-82648f54-6a74-406f-a128-8d4bc7744770.png)